heartbeat/monitors/active/http/check_test.go

```diff
@@ -0,0 +1,106 @@
+package http
+
+import (
+	"fmt"
+	"log"
+	"net/http"
+	"net/http/httptest"
+	"testing"
+
+	"github.com/elastic/beats/libbeat/common/match"
+)
+
+func TestCheckBody(t *testing.T) {
+
+	var matchTests = []struct {
+		description string
+		body        string
+		patterns    []match.Matcher
+		result      bool
+	}{
+		{
+			"Single regex that matches",
+			"ok",
+			[]match.Matcher{match.MustCompile("ok")},
+			true,
+		},
+		{
+			"Regex matching json example",
+			`{"status": "ok"}`,
+			[]match.Matcher{match.MustCompile(`{"status": "ok"}`)},
+			true,
+		},
+		{
+			"Regex matching first line of multiline body string",
+			`first line
+			second line`,
+			[]match.Matcher{match.MustCompile("first")},
+			true,
+		},
+		{
+			"Regex matching lastline of multiline body string",
+			`first line
+			second line`,
+			[]match.Matcher{match.MustCompile("second")},
+			true,
+		},
+		{
+			"Regex matching multiple lines of multiline body string",
+			`first line
+			second line
+			third line`,
+			[]match.Matcher{match.MustCompile("(?s)first.*second.*third")},
+			true,
+		},
+		{
+			"Regex not matching multiple lines of multiline body string",
+			`first line
+			second line
+			third line`,
+			[]match.Matcher{match.MustCompile("(?s)first.*fourth.*third")},
+			false,
+		},
+		{
+			"Single regex that doesn't match",
+			"ok",
+			[]match.Matcher{match.MustCompile("notok")},
+			false,
+		},
+		{
+			"Multiple regex match where at least one must match",
+			"ok",
+			[]match.Matcher{match.MustCompile("ok"), match.MustCompile("yay")},
+			true,
+		},
+		{
+			"Multiple regex match where none of the patterns match",
+			"ok",
+			[]match.Matcher{match.MustCompile("notok"), match.MustCompile("yay")},
+			false,
+		},
+	}
+
+	for _, test := range matchTests {
+		t.Run(test.description, func(t *testing.T) {
+			ts := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
```

Comment:
Requiring an http server is somewhat overkill. Makes me wonder if we can 'simplify' the validators interface/function, so to not depend on a http response.

Especially the fact the `body` validator consuming the complete body is not useful if we want to be able to combine (and/or) more complicated filtering in the future.

Using interfaces one could define:

```
type response struct {
  Status() uint
  Body() ([]byte, error)
  ...
}
```

The adapter for use with http.Response could lazily read and cache the body then.

Reply:
I totally agree 👍 

> Makes me wonder if we can 'simplify' the validators interface/function, so to not depend on a http response.

The wording you used here makes this comment sound like "something we would like to improve in the future". Can you confirm that I interpreted this correctly? Otherwise I'd be happy to have a go refactoring in your suggestion but I would definitely need a bit of help since this goes beyond my basic go knowledge.